### PR TITLE
fix(v3): updates dependencies and unit tests

### PR DESCRIPTION
**Pull request by israx**
Please make sure to read the Pull Request Guidelines:
https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#pull-requests
-->
#### System Requirements
- Node 16.14.0

#### Description of changes

 - ignores deprecated UI packages from lerna to allow building the library
 - fixes `auth`, `predictions` `graph-ql` and `datastore` unit tests

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api-graphql/__tests__/GraphQLAPI-test.ts [hunk 1]
```diff
@@ -757,7 +757,7 @@ describe('API test', () => {
 					variables,
 					authMode: GRAPHQL_AUTH_MODE.OPENID_CONNECT,
 				})
-			).rejects.toThrowError('No federated jwt');
+			).rejects.toThrowError('No current user');
```
Comment (jimblanc) [inline packages/api-graphql/__tests__/GraphQLAPI-test.ts:760]: Sanity check: What did this change? Shouldn't the behavior be the same?

File: packages/auth/src/Auth.ts [hunk 2]
```diff
@@ -1386,7 +1386,9 @@ export class AuthClass {
 		logger.debug('Getting current session');
 		// Purposely not calling the reject method here because we don't need a console error
 		if (!this.userPool) {
-			return Promise.reject();
```
Comment (jimblanc) [inline packages/auth/src/Auth.ts:1389]: Why did we need to change this?

**Review by jimblanc — COMMENTED** [review]
Looks pretty good, thank you! Holding off on approving until we confirm the strategy with the UI packages.

One other validation idea we might want to consider: Would it be possible to diff the artifacts produced by this build with the artifacts available on the latest version of v3 on NPM? This would help us ensure behavior remains the same in lieu of complete E2E tests.

File: packages/auth/src/Auth.ts (hunk 2 above)
Comment (israx) [reply packages/auth/src/Auth.ts:1389]: most APIs, if not all, will construct the same error when a user pool is not configured, and the test is currently asserting for that specific error.

File: packages/api-graphql/__tests__/GraphQLAPI-test.ts (hunk 1 above)
Comment (israx) [reply packages/api-graphql/__tests__/GraphQLAPI-test.ts:760]: Seems the underlying error message changed from `No federated jwt` to `No current user`.

File: .yarnrc [hunk 3]
```diff
@@ -1,2 +1,2 @@
---add.no-lockfile true
---install.no-lockfile true
\ No newline at end of file
+--add.no-lockfile false
```
Comment (ashika112) [inline .yarnrc:1]: I think it is safe to remove this file if we dont need the override. The default behavior should be to add lock-file

File: packages/auth/__tests__/auth-unit-test.ts [hunk 4]
```diff
@@ -3257,7 +3257,7 @@ describe('auth unit test', () => {
 				Username: 'username',
 				Pool: userPool,
 			});
-			const errorMessage = new NoUserPoolError(AuthErrorTypes.EmptyCode);
+			const errorMessage = new NoUserPoolError(AuthErrorTypes.MissingAuthConfig);
```
Comment (ashika112) [inline packages/auth/__tests__/auth-unit-test.ts:3260]: why this change?

File: package.json [hunk 5]
```diff
@@ -36,7 +36,22 @@
 	},
 	"workspaces": {
 		"packages": [
-			"packages/*"
+			"packages/amazon-cognito-identity-js",
```
Comment (ashika112) [inline package.json:39]: Why are we making it explicit now?

Comment (ashika112) [reply package.json:39]: NVM got it . UI ones

File: package.json [hunk 6]
```diff
@@ -91,14 +106,20 @@
 		"webpack": "^4.32.0",
 		"webpack-bundle-analyzer": "^3.3.2",
 		"webpack-cli": "^3.1.0",
-		"winston": "^3.2.1"
+		"winston": "^3.2.1",
+		"@types/lodash": "4.14.182"
```
Comment (ashika112) [inline package.json:110]: can u also explain need for the extra package here?

File: packages/auth/__tests__/auth-unit-test.ts (hunk 4 above)
Comment (israx) [reply packages/auth/__tests__/auth-unit-test.ts:3260]: The test case is checking for an incorrect auth configuration. This test was using a wrong `AuthErrorTypes`

File: package.json (hunk 6 above)
Comment (israx) [reply package.json:110]: It is to help the library to build. V4 currently added this dev dep

File: packages/auth/src/Auth.ts (hunk 2 above)
Comment (israx) [reply packages/auth/src/Auth.ts:1389]: Also the test is checking for no userPool error as well

Comment (jimblanc) [reply packages/auth/src/Auth.ts:1389]: Makes sense, I'm just wondering why the error behavior changed?

**Review by jimblanc — COMMENTED** [review]
Looking good, just had a few more questions about changes to `Auth.ts`, ideally we'd be able to get by here without changing any of the implementation. Also, it seems like this branch kicked off some old workflows which are failing. Do we need to disable those? It looks like they run any time someone adds a comment to this PR 😅

File: packages/auth/src/Auth.ts (hunk 2 above)
Comment (israx) [reply packages/auth/src/Auth.ts:1389]: yeah I don't have an answer respect to that, there is inconsistency between the tests and the code. 

We can either:
1. change/delete the test to match the code.
2. change the code to match the test case.

I would go for number 2 as it is not breaking and is consistent how the other APIs handle the same error when there is no userpool.

**Review by jimblanc — APPROVED** [review]
Discussed concerns async, LGTM. If you haven't yet could you please also make sure the branch can be published using Verdaccio and try installing it in a sample app?
